conf/schemas/dataSets.schema.js

```diff
@@ -0,0 +1,79 @@
+db.runCommand({
+  collMod: "dataSets",
+  validator: {
+    $and: [
+      {
+        dataStoreInfo: { $type: "object", $exists: true },
+      },
+      { "dataStoreInfo.name": { $regex: "^[A-Za-z0-9-_.]+$", $exists: true } },
+      {
+        "dataStoreInfo.url": {
+          $regex: "^https?:/{2}[a-z0-9.-]+(.[a-z]{2,3})?(:[0-9]+)?$",
+          $exists: true,
+        },
+      },
+      { "dataStoreInfo.typ": { $in: ["webknossos-store", "ndstore"], $exists: true } },
+      {
+        $or: [
+          { "dataStoreInfo.accessToken": { $type: "string" } },
+          { "dataStoreInfo.accessToken": { $exists: false } },
+        ],
+      },
+      {
+        dataSource: { $type: "object", $exists: true },
+      },
+      {
+        "dataSource.id": { $type: "object", $exists: true },
+      },
+      {
+        "dataSource.id.name": { $regex: "^[A-Za-z0-9-_.]+$", $exists: true },
+      },
+      {
+        "dataSource.id.team": { $type: "string", $exists: true },
+      },
+      {
+        $or: [
+          { "dataSource.dataLayers": { $type: "object" } },
+          { "dataSource.dataLayers": { $exists: false } },
+          { "dataSource.dataLayers": { $size: 0 } },
+        ],
+      },
+      {
+        $or: [
+          { "dataSource.scale": { $type: "number" } },
+          { "dataSource.scale": { $exists: false } },
+          { "dataSource.scale": { $size: 0 } },
```

Comment:
What is the `size: 0` for?

Reply:
In tests, there were several arrays, which were existent but don't have any content. This leads to the failure of the type check.
One could write an evolution to delete those arrays, but I think the scala code expects an array to return from the db. What's your take on erasing those arrays @jfrohnhofen ?

Reply:
The `scale` property should always exist and have exactly 3 values, so I suggest replacing the `$or` with `$and` and adding `$size: 3` and `$exists: true`. if there are datasets not matching this, they are probably wrong.

Reply:
Ok I will run tests with the new validation and report which datasets fail because I think there were some